### PR TITLE
feat(proxy-path): add proxy path auto-detection with validation

### PR DESCRIPTION
**Pull request by miguelramos**
Implement deriveProxyPath(), normalizeProxyPath(), and validateUniqueProxyPaths() for multi-spec proxy path resolution.

- deriveProxyPath(): explicit priority, fallback to servers[0].url path extraction Returns DeriveProxyPathResult with proxyPath and proxyPathSource (auto|explicit)
- normalizeProxyPath(): leading slash, no trailing slash, reject root /
- validateUniqueProxyPaths(): duplicate + overlap (prefix nesting) detection
- 47 unit tests covering all edge cases

Refs: vite-qq9.4

## Summary by CodeRabbit

* **New Features**
  * Automatic proxy-path detection from OpenAPI server entries with explicit override support.
  * Path normalization and validation to reject too-broad, duplicate, or overlapping proxy routes.

* **Refactor**
  * Public APIs added for proxy-path derivation/normalization/validation and spec ID utilities; exported types updated.

* **Tests**
  * Extensive unit and integration tests covering derivation, normalization, error cases, and uniqueness validation.

### CODE REVIEW

File: .changesets/feature-vite-qq9.4-proxy-path-auto-detection.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  "environments": [
+    "production"
+  ],
+  "packages": [
+    "@websublime/vite-plugin-open-api-server"
+  ],
+  "changes": [],
+  "created_at": "2026-02-18T10:39:32.592010Z",
+  "updated_at": "2026-02-18T10:39:32.592830Z"
+}
```
[the remark this answers is not in the document]
Comment (miguelramos) [reply .changesets/feature-vite-qq9.4-proxy-path-auto-detection.json:46]: Changeset is automatic tool
